### PR TITLE
Fix regression introduced during previous scanner hardening

### PR DESCRIPTION
**Pull request by henryju**
Module workDir is deleted when cleaning root module work dir when they are nested.

### CODE REVIEW

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/WorkDirectoriesInitializer.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      Iterator<Path> it = stream.iterator();
+      while (it.hasNext()) {
+        FileUtils.deleteQuietly(it.next().toFile());
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to clean working directory: " + dirToDelete.toString(), e);
+    }
+  }
+
+  private DirectoryStream<Path> list(Path dir) throws IOException {
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/WorkDirectoriesInitializer.java]: MAJOR Make "list" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/WorkDirectoriesInitializer.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  public void execute() {
+    cleanAllWorkingDirs(moduleHierarchy.root());
+    mkdirsAllWorkingDirs(moduleHierarchy.root());
+  }
+
+  private void cleanAllWorkingDirs(DefaultInputModule module) {
+    for (DefaultInputModule sub : moduleHierarchy.children(module)) {
+      cleanAllWorkingDirs(sub);
+    }
+    if (Files.exists(module.getWorkDir())) {
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/WorkDirectoriesInitializer.java]: MAJOR Replace this with a call to the "toFile().exists()" method [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3725)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/WorkDirectoriesInitializer.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      deleteAllRecursivelyExceptLockFile(module.getWorkDir());
+    }
+  }
+
+  private void mkdirsAllWorkingDirs(DefaultInputModule module) {
+    for (DefaultInputModule sub : moduleHierarchy.children(module)) {
+      mkdirsAllWorkingDirs(sub);
+    }
+    try {
+      org.apache.commons.io.FileUtils.forceMkdir(module.getWorkDir().toFile());
```
Comment (dbmeneses) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/WorkDirectoriesInitializer.java]: Is there an advantage compared to Files.createDirectories?

Comment (henryju) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/WorkDirectoriesInitializer.java]: I don't think so. I'll change.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**